src/ai/ai.cpp

```diff
@@ -667,12 +667,13 @@ void take_ai_decisions(sys::state& state) {
 
 		ve::execute_serial_fast<dcon::nation_id>(state.world.nation_size(), [&](auto ids) {
 			ve::vbitfield_type filter_a = potential
-				? (ve::compress_mask(trigger::evaluate(state, potential, trigger::to_generic(ids), trigger::to_generic(ids), 0)) & !state.world.nation_get_is_player_controlled(ids))
+				? (ve::compress_mask((!potential || trigger::evaluate(state, potential, trigger::to_generic(ids), trigger::to_generic(ids), 0))) & !state.world.nation_get_is_player_controlled(ids))
 				: !state.world.nation_get_is_player_controlled(ids);
 
 			if(filter_a.v != 0) {
+				// empty allow assumed to be an "always = yes"
 				ve::mask_vector filter_c = allow
-					? (trigger::evaluate(state, allow, trigger::to_generic(ids), trigger::to_generic(ids), 0) && (state.world.nation_get_owned_province_count(ids) != 0)) && filter_a
+					? ((!allow || trigger::evaluate(state, allow, trigger::to_generic(ids), trigger::to_generic(ids), 0)) && (state.world.nation_get_owned_province_count(ids) != 0)) && filter_a
```

Comment:
same thing here

Reply:
the real problem is the second validity check, below

Reply:
and then don't forget to make sure that the player is also allowed to take the decision in the command code, if you haven't already